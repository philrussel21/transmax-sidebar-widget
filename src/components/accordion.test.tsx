import {render, fireEvent} from '@testing-library/react';
import Accordion from './accordion';

test('snapshot', () => {
  const result = render(
    <Accordion.Container>
      <Accordion.Panel label="Foo">
        <p>Fooby</p>
      </Accordion.Panel>
      <Accordion.Panel label="Bar">
        <p>Barby</p>
      </Accordion.Panel>
    </Accordion.Container>
  );

  expect(result).toMatchSnapshot();
});

test('it should not have panels displayed by default', () => {
  const {queryByText} = render(
    <Accordion.Container>
      <Accordion.Panel label="Foo">
        <p>Fooby</p>
      </Accordion.Panel>
      <Accordion.Panel label="Bar">
        <p>Barby</p>
      </Accordion.Panel>
    </Accordion.Container>
  );

  expect(queryByText('Fooby')).toBeNull();
  expect(queryByText('Barby')).toBeNull();
});

test('it should display the associated panel when clicked', () => {
  const {getByText, queryByText} = render(
    <Accordion.Container>
      <Accordion.Panel label="Foo">
        <p>Fooby</p>
      </Accordion.Panel>
      <Accordion.Panel label="Bar">
        <p>Barby</p>
      </Accordion.Panel>
    </Accordion.Container>
  );

  fireEvent.click(getByText('Bar'));

  expect(queryByText('Fooby')).toBeNull();
  expect(queryByText('Barby')).not.toBeNull();
});

test('it should display all associated panels when clicked', () => {
  const {getByText, queryByText} = render(
    <Accordion.Container>
      <Accordion.Panel label="Foo">
        <p>Fooby</p>
      </Accordion.Panel>
      <Accordion.Panel label="Bar">
        <p>Barby</p>
      </Accordion.Panel>
    </Accordion.Container>
  );

  fireEvent.click(getByText('Foo'));
  fireEvent.click(getByText('Bar'));

  expect(queryByText('Fooby')).not.toBeNull();
  expect(queryByText('Barby')).not.toBeNull();
});
