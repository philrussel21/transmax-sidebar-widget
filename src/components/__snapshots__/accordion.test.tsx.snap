// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="space-y-4"
      >
        <article
          class="rounded-t-sm bg-[#222931]"
        >
          <header>
            <button
              class="w-full flex justify-between items-center px-4 py-2 border border-white/10"
              type="button"
            >
              <span
                class="uppercase"
              >
                Foo
              </span>
              <svg
                class="size-4 transition "
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                />
              </svg>
            </button>
          </header>
        </article>
        <article
          class="rounded-t-sm bg-[#222931]"
        >
          <header>
            <button
              class="w-full flex justify-between items-center px-4 py-2 border border-white/10"
              type="button"
            >
              <span
                class="uppercase"
              >
                Bar
              </span>
              <svg
                class="size-4 transition "
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                />
              </svg>
            </button>
          </header>
        </article>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="space-y-4"
    >
      <article
        class="rounded-t-sm bg-[#222931]"
      >
        <header>
          <button
            class="w-full flex justify-between items-center px-4 py-2 border border-white/10"
            type="button"
          >
            <span
              class="uppercase"
            >
              Foo
            </span>
            <svg
              class="size-4 transition "
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
              />
            </svg>
          </button>
        </header>
      </article>
      <article
        class="rounded-t-sm bg-[#222931]"
      >
        <header>
          <button
            class="w-full flex justify-between items-center px-4 py-2 border border-white/10"
            type="button"
          >
            <span
              class="uppercase"
            >
              Bar
            </span>
            <svg
              class="size-4 transition "
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
              />
            </svg>
          </button>
        </header>
      </article>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
